lib/class-wp-rest-widget-types-controller.php

```diff
@@ -0,0 +1,456 @@
+<?php
+/**
+ * REST API: WP_REST_Widget_Types_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since x.x.0
+ */
+
+/**
+ * Core class used to access widget types via the REST API.
+ *
+ * @since x.x.0
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Widget_Types_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since x.x.0
+	 */
+	public function __construct() {
+		$this->namespace = '__experimental';
+		$this->rest_base = 'widget-types';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since x.x.0
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_items' ),
+					'permission_callback' => array( $this, 'get_items_permissions_check' ),
+					'args'                => $this->get_collection_params(),
+				),
+				'schema' => array( $this, 'get_public_item_schema' ),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/(?P<name>[a-zA-Z0-9_-]+)',
```

Comment:
Maybe make this available at `/widget/<name>/form-renderer` to be similar to `block-renderer`? I think it just being the `POST` request at the widget endpoint itself is a bit too broad.

Reply:
I had it with way before, change to this. But you are right. I changed it back. https://github.com/WordPress/gutenberg/pull/26042/commits/7d3884e82dbe50581da253ebb95580046e096d20